scoring/score.py

```diff
@@ -1,14 +1,149 @@
+import collections
+
+NUM_TOKENS_OF_EACH_COLOUR = 10
+ROBOT_MULTIPLIER = 3
+
+
 class InvalidScoresheetException(Exception):
     pass
 
 
 class Scorer(object):
+    # Note: if the arena colours (from arena.yaml) change,
+    # both this and the printed score-sheets will need to change
+    ZONE_COLOURS = [
+        'G',    # zone 0 = green
+        'O',    # zone 1 = orange
+        'P',    # zone 2 = purple
+        'Y',    # zone 3 = yellow
+    ]
+
+    ZONE_SCORE_MAP = [
+        [ 2,  2,  2,  2,  2],
+        [ 2,  7,  7,  7,  2],
+        [ 2,  7, 30,  7,  2],
+        [ 2,  7,  7,  7,  2],
+        [ 2,  2,  2,  2,  2],
+    ]
+
     def __init__(self, teams_data, arena_data):
         self._teams_data = teams_data
-        self._arena_data = arena_data
+        self._zone_contents = arena_data['other']['zone_contents']
```

Comment:
Is it ok that we're modifying the original data here?

Reply:
Yes, I think so. SRComp doesn't do anything with the arena data other than pass it to the scorer, so this seems reasonable.
A possible improvement might be to use [`copy.deepcopy`](https://docs.python.org/2.7/library/copy.html#copy.deepcopy) first, but I'm not sure it's worth it (and there likely _is_ a performance penalty for that).